kbase-extension/static/kbase/js/common/cellComponents/filePathWidget.js

```diff
@@ -151,15 +151,33 @@ define([
             // put that in the filePathWidget's data model, then push all the values up the main
             // params bus.
             fieldWidget.bus.on('changed', (message) => {
-                dataModel.rows[rowId].values[parameterSpec.id] = message.newValue;
+                const newValue = message.newValue;
```

Comment:
Every time a value is "changed", it has been successfully validated.
Unfortunately, each widget doesn't know what duplicates are. And having each widget send a bus message up to request the set of duplicates adds complexity, especially if non-changed widgets should show an error.

Instead, when a value is changed, we just update all of our widgets with duplicate errors, if necessary.